courses/views.py

```diff
@@ -107,23 +109,37 @@ def edit_module(request, course_id, module_id=None):
         'action_word': 'Add',
         'form': form
     })
+
 def take_quiz(request, quiz_id):
     quiz = Quiz.objects.get(id=quiz_id)
-    questions=quiz.question_set.all()    
+    # Check if Learner already passed the quiz
+    if (QuizResult.objects.filter(learner_id=request.user.id).exists()
```

Comment:
I think we should be saving a QuizResult even if the user didn't pass the quiz. That way at least we can record their score and maybe in the future we will need to have a history of what the learner has done, how many times they took the quiz, etc.